### ISSUE TITLE
Spurious WPS407 warning reported

### ISSUE THREAD

**Issue by agatti** [wemake-services/wemake-python-styleguide#1624]
# Bug report

`WPS407` warnings are shown for code that arguably follows what the guideline says.

## What's wrong

The following snippet:

```python

from typing import Any, Mapping

MAPPING: Mapping[str, Any] = {}
```

triggers the warning `WPS407 Found mutable module constant`.  As far as I know `Mapping` is the abstract base class for non-mutable dictionaries, so the warning should not be there in the first place. 

## How is that should be

I think that warning should not show up...  Is the snippet above incorrect in defining a non-mutable dictionary in a way that `flake8` won't complain about?

**Comment by sobolevn** [comment]
Well, that's confusing indeed! 

We don't assume types here. Because we cannot really know what `Mapping` in this context really is.
So, intead we look at the value. And the value is mutable. 

Moreover, you can still mutate it with `# type: ignore` or somewhere under `Any`. 
I love to have double guarantees! So, please, use `types.MappingProxyType` or use `noqa` for this case.

Thanks a lot for raising this! 👍
